src/wp-includes/block-template.php

```diff
@@ -224,37 +223,16 @@ function get_the_block_template_html() {
 		return;
 	}
 
-	$content = $wp_embed->run_shortcode( $_wp_current_template_content );
-	$content = $wp_embed->autoembed( $content );
-	$content = shortcode_unautop( $content );
-	$content = do_shortcode( $content );
+	$content = $_wp_current_template_content;
 
-	/*
-	 * Most block themes omit the `core/query` and `core/post-template` blocks in their singular content templates.
-	 * While this technically still works since singular content templates are always for only one post, it results in
-	 * the main query loop never being entered which causes bugs in core and the plugin ecosystem.
+	/**
+	 * Filters the block template HTML content.
 	 *
-	 * The workaround below ensures that the loop is started even for those singular templates. The while loop will by
-	 * definition only go through a single iteration, i.e. `do_blocks()` is only called once. Additional safeguard
-	 * checks are included to ensure the main query loop has not been tampered with and really only encompasses a
-	 * single post.
+	 * @since 6.4.0
 	 *
-	 * Even if the block template contained a `core/query` and `core/post-template` block referencing the main query
-	 * loop, it would not cause errors since it would use a cloned instance and go through the same loop of a single
-	 * post, within the actual main query loop.
+	 * @param string $content The entire block template HTML content.
 	 */
-	if ( is_singular() && 1 === $wp_query->post_count && have_posts() ) {
-		while ( have_posts() ) {
-			the_post();
-			$content = do_blocks( $content );
-		}
-	} else {
-		$content = do_blocks( $content );
-	}
```

Comment:
Instead of changing `do_blocks`, just copy this snippet into a new util function.